service/controller/v14/adapter/adapter.go

```diff
@@ -138,8 +140,11 @@ func NewHostPre(cfg Config) (Adapter, error) {
 func NewHostPost(cfg Config) (Adapter, error) {
 	a := Adapter{}
 
+	a.ClusterID = key.ClusterID(cfg.CustomObject)
```

Comment:
So this adapter pattern and how it is implemented is a bit odd. We started refactoring that a bit so the adapters are actually encapsulating their own functionality. Check for instance the output adapter or instance adapater above. IMO the `Adapter` type here should rather be a collection of adapters that are executed against a given configuration and then applied against their own specific template. When you change or drop an adapter it should not result into other adapters being affected and templates being broken other then the one the adapter itself is responsible for. There should also not be any global variables but each adapter should rather configure itself accordingly. If every adapter needs to deal with the cluster ID, so be it.

Reply:
You mean both `hostRouteTablesAdapter` and `recordSetsAdapter` (and many more) should have `ClusterID` field? Should all of them override this field in `a.get*`? Is is good idea to embed many structs with the same field? Is it even allowed by the compiler? Note that it isn't consistent at all - check `func NewGuest(cfg Config)`.

I don't even want to think about all of those and invest time in that. I'd rather save time here to invest more in https://github.com/giantswarm/giantswarm/issues/3783.

Reply:
IMO the adapter pattern is pretty useless if everything depends on everything and everything accesses global information. So this is why there was already some refactoring done. As I said, check the outputs adapter or the instance adapter of the main stack.

Reply:
Either I don't understand or you want `ClusterID` to be a field of specific `*Adapter` struct.

Then I ask again: do you mean both `hostRouteTablesAdapter` and `recordSetsAdapter` should have `ClusterID` field?

Reply:
I already said yes. When they need it they have to do what they need. This is the point. Every adapter sorts itself and encapsulates its own business logic. So when you rip it out, the others don't bother.

Reply:
Then please also answer my other questions

Reply:
Which one was the other?

Reply:
Should all of them override this field in a.get*? Is is good idea to embed many structs with the same field? Is it even allowed by the compiler? Note that it isn't consistent at all - check func NewGuest(cfg Config).

Reply:
> As I said, check the outputs adapter or the instance adapter of the main stack.

Reply:
This will fail in runtime https://play.golang.org/p/ZX8IzIO-rUi
Beside being flaw in the design this is also confusing and against DRY principle.

Reply:
You did not check the outputs adapter above. This is not how it is in the refactored version. This is not what I mean.

Reply:
I checked yesterday but I didn't get it. I check it now and I understand you want to have field instead of embedding. I agree this is better but I still don't understand why this is blocker for this PR.

Reply:
Nothing is a real blocker. Question is what is important to us and do we want to live with this and does the way we work contribute to our all well being in the future and do we want to invest time now or later. It is your PR. It is your responsibility. Do what you think is right. I can just try to express what I think is right and see that we may get to an agreement that puts us better off together. I do not have to approve everything though. I am also not the only one here able to approve PRs.

Reply:
If this is a blocker I'd like to do it in another PR. I invested a lot of time in testing. Also because of splitting it into two PRs. This refactoring can be easily covered by E2E.

Reply:
Sure, just create issues for the things you want to address later.

Reply:
I will do it right away after merging.

Reply:
Ok cool.

Reply:
https://github.com/giantswarm/giantswarm/issues/3783